packages/lib/src/vis/utils.ts

```diff
@@ -316,17 +318,64 @@ export function getUniforms(
   );
 }
 
-export function getCameraFOV(camera: Camera): {
+export function toArray(arr: NumArray): number[] {
+  return isTypedArray(arr) ? [...arr] : arr;
+}
+
+export function grabCameraScale(camera: Camera): Vector2 {
+  const { scale } = camera;
+  return new Vector2(scale.x, scale.y);
+}
+
+export function grabHtmlPoint(
+  camera: Camera,
+  context: AxisSystemContextValue,
+  dataPt: Vector2 | Vector3
+): Vector2 {
+  const { dataToWorld, cameraToHtml } = context;
+  const worldPt = dataToWorld(dataPt);
+  const cameraPt = new Vector3(worldPt.x, worldPt.y, 0).project(camera);
+  return cameraToHtml(cameraPt);
+}
+
+export function grabHtmlSelection(
+  camera: Camera,
+  context: AxisSystemContextValue,
+  startPoint: Vector2,
+  endPoint: Vector2
+): Selection {
+  return {
+    startPoint: grabHtmlPoint(camera, context, startPoint),
+    endPoint: grabHtmlPoint(camera, context, endPoint),
+  };
+}
+
+export function grabWorldFOV(camera: Camera): {
   topRight: Vector3;
   bottomLeft: Vector3;
 } {
-  // Unproject from normalized camera space (-1, -1) to (1, 1) to world space
+  // Unproject normalized camera viewport [(-1, -1), (1, 1)] to world space
```

Comment:
In OpenGL the viewport is the defined in pixels (https://docs.gl/gl4/glViewport), the [-1, 1] space is called normalized device coordinates
```suggestion
  // Unproject normalized device coordinates [(-1, -1), (1, 1)] to world space
```

Reply:
Hmm, I was trying to explain a bit more clearly that we're computing the world coordinates of the area that the camera is pointing at. Unprojecting from one space to another does not explain why we unproject specifically `CAMERA_TOP_RIGHT` and `CAMERA_BOTTOM_LEFT`, but perhaps the name of the function is clear enough now and I can leave the initial comment as is? I see why the term viewport might be confusing, for sure.